javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,43 @@
+package java.util.concurrent.atomic
+
+import scala.collection.mutable.ArrayBuffer
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    inner = ArrayBuffer[E](array: _*)
+  }
+
+  private var inner: ArrayBuffer[E] = new ArrayBuffer[E](length)
```

Comment:
An `ArrayBuffer` is overkill, since you don't need to resize it. Use an `Array[E]` instead.